src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotAddOrRemoveVirtualKeywordTests.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Xunit;
+
+namespace Microsoft.DotNet.ApiCompatibility.Tests
+{
+    public class CannotAddOrRemoveVirtualKeywordTests
+    {
+        private static readonly bool IsNetFramework = RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework", StringComparison.OrdinalIgnoreCase);
+        private static string CreateType(string s, params object[] args) => string.Format(@"
+namespace CompatTests {{
+  public{0} First
+  {{
+{1}
+  }}
+}}
+", s, string.Join("\n", args));
+
+        private static CompatDifference[] CreateDifferences(params (DifferenceType dt, string memberId)[] args)
+        {
+            var differences = new CompatDifference[args.Length];
+            for (int i = 0; i < args.Length; i++)
+            {
+                if (args[i].dt == DifferenceType.Removed)
+                {
+                    differences[i] = new CompatDifference(DiagnosticIds.CannotRemoveVirtualFromMember, string.Empty, DifferenceType.Removed, args[i].memberId);
+                }
+                else if (args[i].dt == DifferenceType.Added)
+                {
+                    differences[i] = new CompatDifference(DiagnosticIds.CannotAddVirtualToMember, string.Empty, DifferenceType.Added, args[i].memberId);
+                }
```

Comment:
```suggestion
                differences[i] = new CompatDifference(
                    args[i].dt == DifferenceType.Removed ? 
                        new CompatDifference(DiagnosticIds.CannotRemoveVirtualFromMember, string.Empty, DifferenceType.Removed, args[i].memberId) :
                        new CompatDifference(DiagnosticIds.CannotAddVirtualToMember, string.Empty, DifferenceType.Added, args[i].memberId);
```
NIT: this could be simplified